### ISSUE TITLE
Cassandra Storage libraries are out of date

### ISSUE THREAD

**Issue by ligadata-william** [LigaData/Kamanja#1041]
Our DataStax Cassandra driver is version 2.1.2 but their latest version is 3.0.0. Our Cassandra-Thrift library is 2.0.3 but the latest version of cassandra is 3.5 and our libraries don't work with Cassandra 3.x.y.

Additionally, Cassandra 2.1 and 2.2, which we are compatible with, will lose support on Novemeber 2016 according to their download site http://cassandra.apache.org/download/

Cassandra 3.0.x will lose support November 2017.

**Comment by ligadata-pokuri** [comment]
We can upgrade this in 1.5.0

**Comment by ligadata-pokuri** [comment]
Need to make sure we have latest cassandra client package for Kamanja 1.5.0.

**Comment by ligadata-haitham** [comment]
this issue is fixed in the following branch : 
https://github.com/LigaData/Kamanja/tree/1.4.1-Kamanja-issues-1041

.......
- Dependencies updated to latest versions.
- added an exclude jar statement.
- Medical App successfully ran with Cassandra-3.0.6 as storage.

**Comment by ligadata-william** [comment]
this branch didn't get included in 1.4.1. @ligadata-pokuri is this something we want to include in 1.4.1 or should we switch it to 1.5.0?

**Comment by ligadata-pokuri** [comment]
this we can postpone to 1.5.0.

**Comment by ligadata-william** [comment]
While attempting to use cassandra with the latest libraries, I ran into the following exception when trying to add metadata:
```
ERROR [main] - Failed to connect to Datastore
java.lang.ExceptionInInitializerError
	at com.ligadata.keyvaluestore.CassandraAdapter.<init>(CassandraAdapter.scala:215) ~[KamanjaInternalDeps_2.11-1.5.0.jar:0.1-SNAPSHOT]
	at com.ligadata.keyvaluestore.CassandraAdapter$.CreateStorageAdapter(CassandraAdapter.scala:1447) ~[KamanjaInternalDeps_2.11-1.5.0.jar:0.1-SNAPSHOT]
	at com.ligadata.keyvaluestore.KeyValueManager$.Get(KeyValueManager.scala:67) ~[KamanjaInternalDeps_2.11-1.5.0.jar:0.1-SNAPSHOT]
	at com.ligadata.MetadataAPI.PersistenceUtils$.GetDataStoreHandle(PersistenceUtils.scala:779) ~[KamanjaInternalDeps_2.11-1.5.0.jar:0.1-SNAPSHOT]
	at com.ligadata.MetadataAPI.PersistenceUtils$.OpenDbStore(PersistenceUtils.scala:797) [KamanjaInternalDeps_2.11-1.5.0.jar:0.1-SNAPSHOT]
	at com.ligadata.MetadataAPI.MetadataAPIImpl$.OpenDbStore(MetadataAPIImpl.scala:1627) [KamanjaInternalDeps_2.11-1.5.0.jar:0.1-SNAPSHOT]
	at com.ligadata.MetadataAPI.MetadataAPIImpl$.InitMdMgrFromBootStrap(MetadataAPIImpl.scala:4270) [KamanjaInternalDeps_2.11-1.5.0.jar:0.1-SNAPSHOT]
	at com.ligadata.MetadataAPI.StartMetadataAPI$.main(StartMetadataAPI.scala:339) [KamanjaInternalDeps_2.11-1.5.0.jar:0.1-SNAPSHOT]
	at com.ligadata.MetadataAPI.StartMetadataAPI.main(StartMetadataAPI.scala) [KamanjaInternalDeps_2.11-1.5.0.jar:0.1-SNAPSHOT]
Caused by: java.lang.IllegalStateException: Detected Guava issue #1635 which indicates that a version of Guava less than 16.01 is in use.  This introduces codec resolution issues and potentially other incompatibility issues in the driver.  Please upgrade to Guava 16.01 or later.
	at com.datastax.driver.core.SanityChecks.checkGuava(SanityChecks.java:62) ~[ExtDependencyLibs2_2.11-1.5.0.jar:1.0]
	at com.datastax.driver.core.SanityChecks.check(SanityChecks.java:36) ~[ExtDependencyLibs2_2.11-1.5.0.jar:1.0]
	at com.datastax.driver.core.Cluster.<clinit>(Cluster.java:68) ~[ExtDependencyLibs2_2.11-1.5.0.jar:1.0]
	... 9 more
```

Looks like we have yet another Guava dependency issue.

**Comment by ligadata-pokuri** [comment]
Today I will move it to guava 16.0.1

**Comment by ligadata-pokuri** [comment]
Now we are using guava 16.0.1. You can try now.

**Comment by ligadata-pokuri** [comment]
@ligadata-william , can you verify it?

**Comment by ligadata-dhaval** [comment]
Hi, the error says that the guava version should be greater than or equal to 16.01 and we have set it as 16.0.1 instead of 16.01
The dot seems to have caused the confusion. I am updating the build.sbt and trying it again.

**Comment by ligadata-dhaval** [comment]
Hi, I tried 
1.updating the guava version to 19 in build.sbt
2. change sbt version to 0.13.7 and cassandra 3.7
3. Based on the http://datastax.github.io/java-driver/manual/native_protocol/ tried compatibiity check with cassandra 3.0.8 and drivers 3.x
All seem to be giving the same error. Please find the metadata api and cluster config used.

[files .zip](https://github.com/LigaData/Kamanja/files/351038/files.zip)

**Comment by ligadata-william** [comment]
I'm still encountering the guava issue. I don't know why but it seems to not be honoring the dependency override during assembly. I'm not sure why yet.

**Comment by ligadata-william** [comment]
As an additional note: I'm building on OS X. I'll try building on linux to see if something different happens.

**Comment by ligadata-dhaval** [comment]
@ligadata-haitham and I tried out and it seems the issue is with the jars compiled using Mac. Haitham ran the examples on a unix machine and it worked fine. I also used Haitham's binaries and ran it on MAc and it worked fine but if I added my lib folder to his Kamanja it failed. Also tried copying his lib folder into my folder and it worked. There seems some issue with guava on mac while running easyinstall.
